### PR TITLE
test(envoy/bootstrap): add keepalive values to bootstrap test

### PR DESCRIPTION
**Pull request by SanyaKochhar**
Signed-off-by: Sanya Kochhar <kochhars@microsoft.com>

**Description**: Adds values to bootstrap test for keepalive changes made in #5189 

**Testing done**:

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| Tests                      | [x] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? no
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change? no

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)? n/a

### CODE REVIEW

**Review by keithmattix — APPROVED** [review]

**Review by jaellio — APPROVED** [review]
e2es initially failed, but are passing now. Not sure what the flakiness is related to.

File: pkg/envoy/bootstrap/config_test.go [hunk 1]
```diff
@@ -110,6 +110,11 @@ static_resources:
         '@type': type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
         explicit_http_config:
           http2_protocol_options: {}
+    upstream_connection_options:
+      tcp_keepalive:
+        keepalive_interval: 5
+        keepalive_probes: 5
+        keepalive_time: 60
```
Comment (jaellio) [inline pkg/envoy/bootstrap/config_test.go:117]: I believe the `upstream_connection_options` also needs to be added to the `reversedExpectedYAML`.

**Review by jaellio — CHANGES_REQUESTED** [review]

File: pkg/envoy/bootstrap/config_test.go (hunk 1 above)
Comment (keithmattix) [reply pkg/envoy/bootstrap/config_test.go:117]: +1 👍🏾

**Review by steeling — APPROVED** [review]

**Review by keithmattix — CHANGES_REQUESTED** [review]

**Review by jaellio — APPROVED** [review]

**Review by jsturtevant — APPROVED** [review]
Curious why this test didn't fail on the initial PR?

**Review by keithmattix — APPROVED** [review]

**Comment by keithmattix** [comment]
> Curious why this test didn't fail on the initial PR?

It's part of the flaky test I fixed earlier; if we re-ran it a few times before this latest commit, it would've likely failed once

**Comment by jaellio** [comment]
> Curious why this test didn't fail on the initial PR?

Do you mean why weren't the tests failing on this PR https://github.com/openservicemesh/osm/pull/5189?

**Comment by SanyaKochhar** [comment]
Agreed it is super strange, tests were passing when #5189 was merged. Seems like it passed once and failed once on my branch, and same on main after merging. I'm wondering if it's because the config_test expected has two output options and it was matching one of them - https://github.com/openservicemesh/osm/actions/runs/3191116471/jobs/5207109206#step:5:2437

**Pull request merged**